### PR TITLE
OPERATOR-240 [Mastercard] Unable to set image urls when using operator in air gapped environment

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:
https://portworx.atlassian.net/browse/OPERATOR-240

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/util/util_test.go [hunk 1]
```diff
@@ -76,6 +76,21 @@ func TestGetImageURN(t *testing.T) {
...
+	out = GetImageURN("registry.io//", "k8s.gcr.io/pause:3.1")
+	require.Equal(t, "registry.io/pause:3.1", out)
+
+	// Update it again, now k8s.gcr.io should be deleted from common registries.
+	UpdateCommonRegistries([]string{"gcr.io"})
+	out = GetImageURN("registry.io", "k8s.gcr.io/pause:3.1")
+	require.Equal(t, "registry.io/k8s.gcr.io/pause:3.1", out)
+
+	out = GetImageURN("registry.io//", "k8s.gcr.io/pause:3.1")
+	require.Equal(t, "registry.io/pause:3.1", out)
```
Comment (piyush-nimbalkar) [inline pkg/util/util_test.go:90]: Should this have been `registry.io/k8s.gcr.io/pause` as common registries now only has `gcr.io`?

File: pkg/util/util.go [hunk 2]
```diff
@@ -35,8 +35,28 @@ var (
 		"registry-1.docker.io":        true,
 		"registry.connect.redhat.com": true,
 	}
+
+	savedCommonDockerRegistries map[string]bool
 )
 
+// UpdateCommonRegistries add additional registries to common registries.
+func UpdateCommonRegistries(additionalRegistries []string) {
```
Comment (piyush-nimbalkar) [inline pkg/util/util.go:43]: This function will eventually forget about the original common registries and keep on adding to the list.

I think we should avoid saving such a list in utils package in a variable. It is error prone to keep such a singleton util.
I would suggest that in the GetImageURN function we can just pass the list of additional registries (or the whole cluster object and then get the annotation) and replace the common registry if present in the additional list on top of the common registries.

File: pkg/util/util_test.go (hunk 1 above)
Comment (haibinxie) [reply pkg/util/util_test.go:90]: With double slashes as suffix in customRegistry (registry.io//), it always replace the rep, it does not look at common registries, this was old behavior.

File: pkg/util/util.go (hunk 2 above)
Comment (haibinxie) [reply pkg/util/util.go:43]: Sounds good, I can make it as another parameter of GetImageURN

File: pkg/util/util_test.go (hunk 1 above)
Comment (piyush-nimbalkar) [reply pkg/util/util_test.go:90]: Oh right, I forgot that part.

**Comment by haibinxie** [comment]
@piyush-nimbalkar I resolved your comments.

File: pkg/util/util_test.go [hunk 3]
```diff
@@ -9,74 +10,95 @@ import (
...
 
-	out = GetImageURN("registry.io//", "test/this/image")
+	out = getImageURN("", "registry.io//", "test/this/image")
 	require.Equal(t, "registry.io/image", out)
+
+	out = getImageURN("k8s.gcr.io", "registry.io//", "k8s.gcr.io/pause:3.1")
+	require.Equal(t, "registry.io/pause:3.1", out)
+
+	// Update it again, now k8s.gcr.io should be deleted from common registries.
+	out = getImageURN("gcr.io", "registry.io", "k8s.gcr.io/pause:3.1")
```
Comment (piyush-nimbalkar) [inline pkg/util/util_test.go:86]: Can we add a test for comma-separated multiple registries?

File: pkg/util/util.go [hunk 4]
```diff
@@ -39,12 +40,37 @@ var (
 	}
 )
 
+// UpdateCommonRegistries add additional registries to common registries.
```
Comment (piyush-nimbalkar) [inline pkg/util/util.go:43]: Update the comment

**Review by piyush-nimbalkar — APPROVED** [review]

**Comment by haibinxie** [comment]
@piyush-nimbalkar I resolved your last comments and merged the PR.

**Pull request merged**